src/main/java/Figure.java

```diff
@@ -0,0 +1,46 @@
+public abstract class Figure implements FigureInterface {
+    protected double area;
+    private String name;
+    private String color;
+
+    public Figure() {
+    }
+
+    public Figure(String name) {
+        this.name = name;
+        ColorSupplier colorSupplier = new ColorSupplier();
```

Comment:
let;s define random color in FiguresSupplier class, and accept it as constructor parameter